expression/builtin_like.go

```diff
@@ -80,6 +82,12 @@ func (b *builtinLikeSig) evalInt(row chunk.Row) (int64, bool, error) {
 		return 0, isNull, errors.Trace(err)
 	}
 	escape := byte(val)
+
+	if variable.SysVars["lower_case_table_names"].Value != "0" {
```

Comment:
I am afraid that this would cause inconsistent result between tidb and tikv. See line 68: "NOTE: Currently tikv's like function is case sensitive, so we keep its behavior here."

Reply:
@iamzhoug37 If the `like` function is pushed to tikv, then tikv behaves different from tidb and mysql, because:
- In tidb and mysql: `like` is case insensitive
- In tikv, `like` is case sensitive

I think you should file a pull request in tikv to change the `like` function to be case insensitive as well.

Reply:
/cc @crazycs520

Reply:
@iamxy Can you help to implement case insensitive in TIKV? I'm not familiar with TIKV..

Reply:
@crazycs520 Can you help to implement case insensitive in TIKV? I'm not familiar with TIKV..

Reply:
ok, let me have a try.

Reply:
Thank you !

Reply:
Maybe even if we change the tikv's behavior, we cannot make it right...Because when there is an index, we cannot build scan ranges to make it case insensitive...

Reply:
@zz-jason PTAL 😨